css-viewport-1/Overview.bs

```diff
@@ -526,6 +526,52 @@ The <dfn export>unscaled</dfn> value of a CSS length relative to an element is t
 	The {{Window/devicePixelRatio}} of a frame is multiplied by the [=effective zoom=] inherited by its parent frame.
 </h3>
 
+<h2 id=extensions-to-the-window-interface>Extensions to the {{Window}} Interface</h2>
+
+<pre class=idl>
+partial interface Window {
+[SameObject, Replaceable] readonly attribute Viewport viewport;
+};
+</pre>
+
+<h2 id=viewport>Viewport</h2>
+
+<h3 id="the-viewport-interface">The {{Viewport}} Interface</h3>
+
+<pre class=idl>
+[Exposed=Window]
+interface Viewport {
+  readonly attribute FrozenArray&lt;DOMRect> segments;
+};
+</pre>
+
+<h2 id='segments'>
+	The <dfn attribute for=Viewport>segments</dfn> property
+</h2>
+
+The {{segments}} property is an array of {{DOMRect}} that represent the dimensions of each existing viewport segment.
+Each {{DOMRect}} contains the geometry of the segment (x, y, width, height) in ''<length>/px''.
+
+Additonal details about the definition of a viewport segment can be found here: [[css-env-1#viewport-segments]].
+
+The {{segments}} attribute must run these steps:
+1. If the {{Viewport}}'s associated {{Document}} is not <a>fully active</a>, return 0.
+2. Returns null if there is only a single viewport segment and abort these steps.
```

Comment:
Why not return an array with only 1 segment `{ x:0, y:0, width:window.innerWidth, height:window.innerHeight }` here? Seems more ergonomic for authors than checking for `null`/_falsy_.

Reply:
I answered earlier in the comments. 

>Personally, I can't wrap my head around a single segment. It's either the viewport is segmented or it's not. If it's segmented, then it will be two or more segments.

> Currently in chromium-based browsers the media queries also will not resolve when the viewport is not segmented,@media (vertical-viewport-segments: 1) doesn't work. Also returning a segment that is the exact viewport size doesn't bring any value and that's why the Android native API doesn't give any value when the viewport is not segmented. I feel like developers will start writing code @media (vertical-viewport-segments: 1) or if (window.viewport.segments == 1) which is not what we want, there is no special treatment to be done when the view is not segmented, it's a regular device.

> Overall, we want the developers to think about what to do when the view is segmented, when it's not it is the default behavior (foldable device or not)."

Reply:
Thanks for the clarification.